base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1179 @@
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\RAPPS Downloads");
+    }*/
+
+    SHGetFolderPathW(NULL, CSIDL_PERSONAL, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+    StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\RAPPS Downloads");
+    if (PathIsDirectoryW(TargetedDir))
+    {
+        sz.RappsSize = DirSizeFunc(TargetedDir);
+        if (sz.RappsSize == 1)
+        {
+            return FALSE;
+        }
+        StringCbCopyW(wcv.RappsDir, _countof(wcv.RappsDir), TargetedDir);
+        ZeroMemory(&TargetedDir, sizeof(TargetedDir));
+        SendMessageW(hProgressBar, PBM_SETPOS, 100, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Temporary RAPPS Files");
+    }
+    
+    SendMessageW(hwnd, WM_DESTROY, 0, 0);
+    return TRUE;
+}
+
+DWORD WINAPI FolderRemoval(LPVOID lpParam)
+{
+    HWND hProgressBar = NULL;
+    WCHAR TargetedDir[MAX_PATH] = { 0 };
+    
+    HWND hwnd = (HWND)lpParam;
+
+    hProgressBar = GetDlgItem(hwnd, IDC_PROGRESS_2);
+
+    if (bv.TempClean == TRUE && bv.SysDrive == TRUE)
+    {
+        SHGetFolderPathW(NULL, CSIDL_WINDOWS, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Temp");
+        CleanRequiredPath(TargetedDir);
+        
+        SHGetFolderPathW(NULL, CSIDL_PROFILE, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Local Settings\\Temp");
+        CleanRequiredPath(TargetedDir);
+        
+        SendMessageW(hProgressBar, PBM_SETPOS, 25, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Temporary files");
+    }
+    
+    if (bv.RecycleClean == TRUE)
+    {   
+        StringCbCopyW(TargetedDir, _countof(TargetedDir), wcv.DriveLetter);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\");
+        
+        SHEmptyRecycleBinW(NULL, TargetedDir, SHERB_NOCONFIRMATION | SHERB_NOPROGRESSUI | SHERB_NOSOUND);
+        
+        SendMessageW(hProgressBar, PBM_SETPOS, 50, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Recycled files");
+    }
+    
+    if (bv.ChkDskClean == TRUE)
+    {
+        for(int i = 0; i < 10; i++)
+        {
+            StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s\\FOUND.%.3i", wcv.DriveLetter, i);
+            if (PathIsDirectoryW(TargetedDir))
+            {
+                CleanRequiredPath(TargetedDir);
+            }
+        
+            else
+                break;
+        }
+        SendMessageW(hProgressBar, PBM_SETPOS, 75, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Old ChkDsk files");
+    }
+
+    if (bv.RappsClean == TRUE)
+    {
+        CleanRequiredPath(wcv.RappsDir);
+
+        SendMessageW(hProgressBar, PBM_SETPOS, 100, 0);
+        SetDlgItemTextW(hwnd, IDC_STATIC_INFO, L"Temporary RAPPS Files");
+    }
+    
+    SendMessageW(hwnd, WM_DESTROY, 0, 0);
+
+    return TRUE;
+}
+
+void CleanRequiredPath(PWCHAR TempPath)
```

Comment:
`PCWSTR` is preferable for null-terminated input strings because
* it's a pointer to `const`, indicating your function won't modify the buffer
* the `-STR` typedefs include a null-terminated annotation that helps static analysis.